SPHINXsys/src/shared/particle_dynamics/solid_dynamics/contact_dynamics.cpp

```diff
@@ -129,6 +126,27 @@ namespace SPH
 			}
 		}
 		//=================================================================================================//
+		void ShellContactDensity::interaction(size_t index_i, Real dt)
+		{
+			/** shell contact interaction. */
+			Real sigma = 0.0;
+			Real contact_density_i = 0.0;
+
+			for (size_t k = 0; k < contact_configuration_.size(); ++k)
+			{
+				StdLargeVec<Real>& contact_Vol_k = *(contact_Vol_[k]);
+				Neighborhood& contact_neighborhood = (*contact_configuration_[k])[index_i];
+				for (size_t n = 0; n != contact_neighborhood.current_size_; ++n)
+				{
+					Real corrected_W_ij = std::max(contact_neighborhood.W_ij_[n] - offset_W_ij_[k], 0.0);
+					sigma += corrected_W_ij * contact_Vol_k[contact_neighborhood.j_[n]];
+				}
+				constexpr Real heuristic_limiter = 0.005;
```

Comment:
```suggestion
				constexpr Real heuristic_limiter = 0.005;
				// The maximum contact pressure is the bulk modulus K=density*c*c,  wherein c is the sound wave speed. 
				// The contact pressure is huge and applied on fewer particles than for solids, yielding high acceleration locally, which is one source of instability. 
```

Reply:
Wouldn't it be good to look at whether this value can be replaced by the shell thickness since it seems to be in the same range ?

Reply:
I think the thickness should not only be a multiplier but the dp be replaced by the thickness for calculating contact distance and thus the contact density

Reply:
Yes, the dp should be replaced by thickness in some way. It is important that we need to ensure that both bodies can search for each other's particles as their neighbor particles when updating `contact_configuration_`.